src/openhab.cpp

```diff
@@ -70,100 +66,161 @@ OpenHAB::OpenHAB(const QVariantMap& config, EntitiesInterface* entities, Notific
+                doc = QJsonDocument::fromJson(data.mid(6).left(data.lastIndexOf("}") + 1), &parseerror);
+            }
 
-        QVariantMap map = doc.toVariant().toMap();
-        // only process state changes
-        if ((map.value("type").toString() == "ItemStateEvent") ||
-                (map.value("type").toString() == "GroupItemStateChangedEvent")) {
-            // get item name from the topic string
-            // example: smarthome/items/EG_Esszimmer_Sonos_CurrentPlayingTime/state
-            QString          name = map.value("topic").toString().split('/')[2];
-            EntityInterface* entity = m_entities->getEntityInterface(name);
-            if (entity != nullptr && entity->connected()) {
-                QString value =
-                        (reinterpret_cast<const QVariantMap*>(map.value("payload").data()))->value("value").toString();
-                // because OpenHab doesn't send the item type in the status update, we have to extract it from our own
-                // entity library
-                if (entity->type() == "light" && entity->supported_features().contains("BRIGHTNESS") &&
-                        regex_brightnessvalue.exactMatch(value)) {
-                    processLight(value, entity, true);
-                } else if (entity->type() == "light" && entity->supported_features().contains("COLOR") &&
-                           regex_colorvalue.exactMatch(value)) {
-                    processComplexLight(value, entity);
-                } else if (entity->type() == "light") {
-                    processLight(value, entity, false);
-                } else if (entity->type() == "blind") {
-                    processBlind(value, entity);
-                } else if (entity->type() == "switch") {
-                    processSwitch(value, entity);
+            if (parseerror.error != QJsonParseError::NoError && !_flagMoreDataNeeded) {
+                qCDebug(m_logCategory) << QString(doc.toJson(QJsonDocument::Compact)) << "read " << data.size()
+                                       << "bytes"
+                                       << "read data" << data.mid(6) << "SSE JSON error:" << parseerror.error
+                                       << parseerror.errorString();
+                continue;
+            }
+            // only process state changes
+            if (!_flagMoreDataNeeded) {
+                if ((doc.object().value("type").toString() == "ItemStateEvent") ||
+                    (doc.object().value("type").toString() == "GroupItemStateChangedEvent")) {
+                    // get item name from the topic string
+                    // example: smarthome/items/EG_Esszimmer_Sonos_CurrentPlayingTime/state
+                    QString          name = doc.object().value("topic").toString().split('/')[2];
+                    EntityInterface* entity = m_entities->getEntityInterface(name);
+                    if (entity != nullptr && entity->connected()) {
+                        pyload =
```

Comment:
Use a local variable for pyoad. Doesn't seem to be used outside this method.

Reply:
done